test/query_component_test.dart

```diff
@@ -72,17 +77,137 @@ void main() {
 
   tearDown(disposeAnyRunningTest);
 
-  group('populate button clicks invokes QueryService function:', () {
-    setUp(() async => await queryComponentPO.clickPopulate());
-
-    test('execDV3Query', () => verify(mockQueryService.execDV3Query(any, any)));
-
-    test('execReportingCreateQuery',
-        () => verify(mockQueryService.execReportingCreateQuery(any, any, any)));
+  group('radio button clicks change QueryType:', () {
+    test(
+        'selecting byAdvertiser invokes execDV3Query(Query.byAdvertiser, '
+        ', null, null, null)', () async {
+      await (queryComponentPO.selectByAdvertiser());
+      await (queryComponentPO.clickPopulate());
+      verify(mockQueryService.execDV3Query(
+          QueryType.byAdvertiser, '', null, null, null));
+    });
+
+    test(
+        'selecting byAdvertiser invokes execDV3Query(Query.byAdvertiser, '
+        ', null, null, null)', () async {
+      await (queryComponentPO.selectByMediaPlan());
+      await (queryComponentPO.clickPopulate());
+      verify(mockQueryService.execDV3Query(
+          QueryType.byMediaPlan, '', null, null, null));
+    });
+
+    test(
+        'selecting byAdvertiser invokes execDV3Query(Query.byAdvertiser, '
+        ', null, null, null)', () async {
+      await (queryComponentPO.selectByIO());
+      await (queryComponentPO.clickPopulate());
+      verify(mockQueryService.execDV3Query(
+          QueryType.byInsertionOrder, '', null, null, null));
+    });
+
+    test('only the last selection matters', () async {
+      await (queryComponentPO.selectByIO());
+      await (queryComponentPO.selectByAdvertiser());
+      await (queryComponentPO.selectByMediaPlan());
+      await (queryComponentPO.clickPopulate());
+      verify(mockQueryService.execDV3Query(
+          QueryType.byMediaPlan, '', null, null, null));
+    });
   });
 
-  test('populate button clicks invokes ExcelDart.populate()', () async {
-    await queryComponentPO.clickPopulate();
-    verify(mockExcelDart.populate(any, any));
+  group('populate-button clicks invokes functions:', () {
```

Comment:
maybe a better test would be to execQuery and expect on the output. This way if you change your implementation, only the setup will have to be changed not the test itself. Test behavior not implementation. Is there an output on execQuery we can observe instead?

Note: this is different from testing interaction w/ office APIs b/c we control the implementation here.

Reply:
execQuery depends on google js api to make requests. I tried and it seems like the js wrapper functions cannot be called in testing environment? We could talk more about this.